rbd/rbd.go

```diff
@@ -215,6 +215,27 @@ func Create(ioctx *rados.IOContext, name string, size uint64, order int,
 	}, nil
 }
 
+func CreateWithOptions(ioctx *rados.IOContext, name string, size uint64, rio *RbdImageOptions) (image *Image, err error) {
```

Comment:
Would it make more sense to stick to ceph's current naming convention of create4? The current approach to Create in rdb.go is a big mess IMO and I want to make sure we don't make it worse.

Reply:
I dont know. `Create4()` is not really user-friendly, and people might wonder where `Create2()` and `Create3()` are. But the decision also depends on the goals of the project. If it is the idea to stick to the `librbd.so` API as much as possible, `Create4()` would be more suitable.

Reply:
Yeah, its definitely a value judgement. As I also just started working on the project it feels odd for me to impose anything but I think its worth discussing. When I have worked with wrapper api's in the past those that are "thin" are generally easier to extend and make the most of often at the cost of being a bit less convient. "Thick" apis that put a convenience layer are often easier to use at the start but sometimes they become inconvenience layers if you just want to use the api in ways the layering doesn't support. I most prefer thin apis that have _optional_ thick parts for stuff that common, very non-idomatic for the language, or difficult ot use correctly. However, to avoid getting side tracked too much this one is just about the naming. I am OK with the name you chose as long as you put a godoc style comment indicating this calls the Create4 function in the ceph api. I'm also happy to continue discussing.

Reply:
There definitely is some merit in a thin API for librados.so. I prefer to offer Golang programmers an API that feels natural for them, which of course can can be done in addition to the thin API. I guess that renaming this to `Create4()` and adding the other `Create*()` (in a different PR) addresses your concerns.

Reply:
This is done now, renamed to `Create4()`.

Reply:
Thanks. We can add a more go-idomatic layer at some time later if/when it is appropriate.